src/controller/buffer-controller.ts

```diff
@@ -1026,9 +1026,9 @@ export default class BufferController implements ComponentAPI {
   };
 
   private get mediaSrc(): string | undefined {
-    const media =
-      (this.media?.firstChild as HTMLSourceElement | null) || this.media;
-    return media?.src;
+    if (!this.media) return undefined;
+    const media = getSourceChild(this.media) || this.media;
+    return media.src;
```

Comment:
Can we keep this closer to the original and drop `getSourceChild`?

```suggestion
    const media = this.media?.querySelector?.('source') || this.media;
    return media?.src;
```

Reply:
Yes, done on 13301b49cd4cdd3979b25dd3b232007fcbe12680